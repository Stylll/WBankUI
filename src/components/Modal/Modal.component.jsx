import React from 'react';
import PropTypes from 'prop-types';

import './Modal.styles.scss';

const Modal = (props) => {
  const {
    children, isModalVisible, handleModalClose,
    customContentClass,
  } = props;

  const onModalClose = () => {
    return handleModalClose();
  };

  const renderModal = () => (
    <div className="modal-container">
      <div className="close-btn" data-testid="modal-close" onClick={onModalClose}>
        &times;
      </div>
      <div className={`content ${customContentClass}`}>
        {children}
      </div>
    </div>
  );

  return (
    <div className="modal">
      {isModalVisible && renderModal()}
    </div>
  );
};

Modal.propTypes = {
  handleModalClose: PropTypes.func.isRequired,
  isModalVisible: PropTypes.bool.isRequired,
  children: PropTypes.node,
  customContentClass: PropTypes.string,
};

Modal.defaultProps = {
  children: null,
  isModalVisible: false,
  customContentClass: '',
};

export default Modal;
