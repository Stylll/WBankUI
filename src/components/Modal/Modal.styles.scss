@import '../../partials/colors';

.modal {
  z-index: 150;
  .modal-container {
    z-index: 151;
    width: 100%;
    height: 100%;
    position: fixed;
    background-color: $overlay-color;
    top: 0; 
    left: 0;
    right: 0;
    bottom: 0;
    overflow-y: scroll;
    display: flex;
    justify-content: center;
    align-items: center;
    -ms-overflow-style: none;
    scrollbar-width: none;
    &::-webkit-scrollbar {
      display: none;
    }
    .content {
      background-color: transparent;
      border-radius: 0px;
      margin: auto;
      max-width: 100%;
      width: 100%;
      height: 100vh;
    }
    .close-btn {
      color: $white;
      position: fixed;
      z-index: 4;
      top: 10px;
      right: 10px;
      font-size: 35px;
      cursor: pointer;
      display: flex;
      &:hover {
        color: $gray;
      }
    }
  }
}
