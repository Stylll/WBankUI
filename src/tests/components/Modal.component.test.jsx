import { render } from '@testing-library/react';
import React from 'react';

import Modal from '../../components/Modal/Modal.component';

describe('Modal Component', () => {
  it('should render properly', () => {
    const { baseElement } = render(<Modal
      customContentClass="class"
      handleModalClose={() => {}} />);

    expect(baseElement).toMatchSnapshot();
  });
});
