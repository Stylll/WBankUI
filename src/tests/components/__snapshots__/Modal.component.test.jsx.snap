// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should render properly 1`] = `
<body>
  <div>
    <div
      class="modal"
    />
  </div>
</body>
`;
